core/src/commonJvmMain/kotlin/com/ustadmobile/core/catalog/contenttype/XapiTypePluginCommonJvm.kt

```diff
@@ -1,92 +1,173 @@
+                    title = if (activity.name.isNullOrEmpty())
+                        uri.getFileName(context) else activity.name
+                    contentTypeFlag = ContentEntry.TYPE_INTERACTIVE_EXERCISE
+                    description = activity.desc
+                    leaf = true
+                    entryId = activity.id
+                    sourceUrl = uri.uri.toString()
                 }
-            } catch (e: IOException) {
-                e.printStackTrace()
-            } catch (e: XmlPullParserException) {
-                e.printStackTrace()
+                MetadataResult(entry, PLUGIN_ID)
             }
-
-            contentEntry
         }
     }
 
-    override suspend fun importToContainer(uri: String, conversionParams: Map<String, String>,
-                                           contentEntryUid: Long, mimeType: String,
-                                           containerBaseDir: String, context: Any,
-                                           db: UmAppDatabase, repo: UmAppDatabase,
-                                           progressListener: (Int) -> Unit): Container {
+    override suspend fun processJob(jobItem: ContentJobItemAndContentJob, process: ProcessContext, progress: ContentJobProgressListener): ProcessResult {
+        val contentJobItem = jobItem.contentJobItem ?: throw IllegalArgumentException("mising job item")
+        val uri = contentJobItem.sourceUri ?: return ProcessResult(JobStatus.FAILED)
         return withContext(Dispatchers.Default) {
 
-            val doorUri = DoorUri.parse(uri)
-            val container = Container().apply {
-                containerContentEntryUid = contentEntryUid
-                cntLastModified = System.currentTimeMillis()
-                this.mimeType = mimeType
-                containerUid = repo.containerDao.insert(this)
-            }
+            try {
+
+                val doorUri = DoorUri.parse(uri)
+                val localUri = process.getLocalUri(doorUri, context, di)
+                val trackerUrl = db.siteDao.getSiteAsync()?.torrentAnnounceUrl
+                        ?: throw IllegalArgumentException("missing tracker url")
+                val contentNeedUpload = !doorUri.isRemote()
+                val progressSize = if (contentNeedUpload) 2 else 1
+
+                val container = db.containerDao.findByUid(contentJobItem.cjiContainerUid)
+                        ?: Container().apply {
+                            containerContentEntryUid = contentJobItem.cjiContentEntryUid
+                            cntLastModified = System.currentTimeMillis()
+                            mimeType = supportedMimeTypes.first()
+                            containerUid = repo.containerDao.insertAsync(this)
+                            contentJobItem.cjiContainerUid = containerUid
+                        }
+
+                db.contentJobItemDao.updateContainer(contentJobItem.cjiUid, container.containerUid)
```

Comment:
Why update the container just after creating it? That seems pointless. You could use arguments in addEntriesToCotnainerFromZip to make sure the container is updated (if that is not already done by default)

Reply:
this was updating the contentJobItem containerUid, updated the name